tests/ui/infinite_loop.rs

```diff
@@ -177,6 +177,23 @@ impl Counter {
     }
 }
 
+fn while_loop_with_break_and_return() {
+    let y = 0;
+    while y < 10 {
```

Comment:
We still want to lint this until we split up the lints.

Reply:
I am not sure I follow. Do you want me to ignore `has_break_or_return_visitor.has_break_or_return` until the other lint is done?

Reply:
Hmm... I read your comment
> Next I will try to break this lint into 2 as suggested in #4648 (comment)

like you want to do this in this PR. But if we should merge this PR as an intermediate fix for this lint, the current behavior is fine.

What I actually meant with this was, that this case should still get linted (but by another lint).

So you choose: Either, we can merge this PR as an intermediate fix or you can split up this lint in this PR.

Reply:
Yes, let's merge this one, if you don't mind, and I will start working on the other lint in a separate PR. Thanks.